File: kotlin/src/main/kotlin/357.CountNumberswithUniqueDigits.kt
import org.junit.Assert.assertEquals
import org.junit.Test
//https://leetcode.com/problems/count-numbers-with-unique-digits/


class CountNumberswithUniqueDigits{
  fun countNumbersWithUniqueDigits(n: Int): Int {
    val muls = listOf(9) + (9 downTo 1)
    var cnt = 1
    var mul = 1
    for(i in 0 until minOf(n, muls.size)){
      mul *= muls[i]
      cnt += mul
    }
    return cnt
  }

  @Test
  fun test1(){
    assertEquals(91, countNumbersWithUniqueDigits(2))
  }

  @Test
  fun test2(){
    assertEquals(10, countNumbersWithUniqueDigits(1))
  }
}

File: python/357_CountNumberswithUniqueDigits.py
from unittest import TestCase
# https://leetcode.com/problems/count-numbers-with-unique-digits/


class CountNumberswithUniqueDigits(TestCase):
    def countNumbersWithUniqueDigits(self, n):
        """
        :type n: int
        :rtype: int
        """
        ret, mul = 1, 1
        for k in [9, *range(9, 0, -1)][:n]:
            mul *= k
            ret += mul
        return ret

    def test1(self):
        self.assertEqual(91, self.countNumbersWithUniqueDigits(2))
